### DISCUSSION TITLE
No Xaml Usage

### DISCUSSION THREAD

**Discussion by AathifMahir** [CommunityToolkit/Maui#1184] [General]
Hello,

As far as reviewing some of the codes of communitytoolkit.maui I found that there's literally no usage of Xaml in many places where it makes sense, is there any specific reason to the decision where no usage of Xaml and everything written in C# even though Xaml and C# were same under the hood but writing some UI elements in Xaml makes sense and easier.

**Comment by bijington** [comment]
It can come entirely down to preference, our samples are aimed at covering as many possibilities as we can in order to show different ways of building UIs. @brminnick will also inform you that not using XAML can provide some performance improvements because the XAML compiler doesn't compile everything and therefore we still rely on some runtime overheads.

For more usage scenarios on not using XAML you should checkout our Markup repository:
https://github.com/CommunityToolkit/Maui.Markup

**Reply by TheCodeTraveler** [reply]
I think it’s important to also note that XAML is a tool for developers to consume UI frameworks. It is not a tool that developers use to create UI frameworks. 

For example, you cannot create a custom control that requires a Handler in XAML. However, you can consume that custom control in your app using XAML.

**Reply by AathifMahir** [reply]
> I think it’s important to also note that XAML is a tool for developers to consume UI frameworks. It is not a tool that developers use to create UI frameworks. 
> 
> For example, you cannot create a custom control that requires a Handler in XAML. However, you can consume that custom control in your app using XAML. 

That makes total sense, I'd keep it in mind when developing new packages for maui

**Comment by CliffAgius** [comment]
Closed and answered
